### DISCUSSION TITLE
嗅探到目标链接后，能否以非手动复制方式获取到？

### DISCUSSION THREAD

**Discussion by Aaaandys** [xifangczy/cat-catch#92] [Q&A]
大家好，请问是否有方式当插件嗅探到了链接后，能否自动将链接传到我们指定的地方，或者说请求某个接口能得到嗅探到的所有链接，而不是必须在浏览器中手动复制。（我对这方面不熟悉，可能表达不太好。）

**Comment by xifangczy** [comment]
额... 你要弄到什么地方？

**Reply by Aaaandys** [reply]
你好~ 因为在做一个视频网站网络质量监控相关的系统，就希望能够获取到视频的下载链接，模拟下载操作，获取下载速度之类参数。我期望的整个流程为：用程序模拟浏览器打开视频网站行为，嗅探到目标连接，程序通过某种方式能获取到嗅探地址（比如说嗅探完成后这个地址已经被写入本地某个文件）。

**Reply by xifangczy** [reply]
可以啊，方法很多 https://github.com/xifangczy/cat-catch/blob/272eee883fe9e8a642b53b535c27fb8c47770e4a/js/background.js#L191
这个函数体内 最后一行添加相关代码就可以了。

可以用伪协议 关键词: URL Protocol 
也可以用 https://developer.chrome.com/docs/extensions/reference/downloads/#method-download Chrome的下载API 把链接放到txt里下载

**Reply by Aaaandys** [reply]
非常感谢你的回复😊😊😊！我去研究一下相关实现~
